backend/app/services/search.py

```diff
@@ -97,22 +102,29 @@ async def post(self, query, indice=None, size=10000, start_date=None, end_date=N
                             response = await self.new_es.search(
                                 index=self.new_index+"*",
                                 body=jsonable_encoder(query),
-                                size=size)
-                            new_results = response['hits']['hits']
+                                size=size)   
+                            new_results = {"data":response['hits']['hits'],"total":response['hits']['total']['value']}
                         else:
-                            new_results = await self.scan_indices(self.new_es, self.new_index, query, timestamp_field, new_start_date, end_date, size)
-                    return await self.remove_duplicates(previous_results + new_results)
+                            response = await self.new_es.search(
+                                index=self.new_index+"*",
+                                body=jsonable_encoder(query),
+                                size=size)   
+                            new_results = {"data":response['hits']['hits'],"total":response['hits']['total']['value']}             
+                            # new_results = await self.scan_indices(self.new_es, self.new_index, query, timestamp_field, new_start_date, end_date, size)
+                    unique_data = await self.remove_duplicates(previous_results["data"] if("data" in previous_results) else [] + new_results["data"]  if("data" in new_results) else[])
+                    totalVal = previous_results["total"] if("total" in previous_results) else 0 + new_results["total"]  if("total" in new_results) else 0 
+                    return ({"data":unique_data, "total": totalVal})
                 else:
                     if start_date and end_date:
                         query['query']['bool']['filter']['range'][timestamp_field]['gte'] = str(start_date)
                         query['query']['bool']['filter']['range'][timestamp_field]['lte'] = str(end_date)
-                        return await self.scan_indices(self.new_es, self.new_index, query, timestamp_field, start_date, end_date, size)
-                    else:
+                        # return await self.scan_indices(self.new_es, self.new_index, query, timestamp_field, start_date, end_date, size)
+                    # else:
```

Comment:
I'm concerned that this change might inadvertently impact other scenarios, even though I've verified it against most cases. If it doesn't affect any other cases then these lines can be taken out or it can be reverted.